app/services/teacher_training_public_api/sync_sites.rb

```diff
@@ -112,5 +113,16 @@ def handle_course_options_with_invalid_sites(sites)
         course_option.update!(site_still_valid: false)
       end
     end
+
+    def handle_course_options_with_reinstated_sites(sites)
+      withdrawn_course_options = @course.course_options.joins(:site).where(site_still_valid: false)
+      site_codes = sites.map(&:code)
+
+      course_options_to_reinstate = withdrawn_course_options.select do |course_option|
+        site_codes.include?(course_option.site.code)
+      end
+
+      course_options_to_reinstate.update_all(site_still_valid: true)
```

Comment:
`update_all` skips auditing. Do we want to use `update` instead and maintain that?

Reply:
Yes, and `update_all` didn't work in any case.